packages/core/core/src/services/document-service/transform/utils.ts

```diff
@@ -1,9 +1,72 @@
-const switchIdForDocumentId = (output: Record<string, any>) => {
+import { Common } from '@strapi/types';
+import { traverse } from '@strapi/utils';
+
+import { IdMap } from './id-map';
+
+export const switchIdForDocumentId = (output: Record<string, any>) => {
   // Mutating for performance reasons
   const documentId = output.documentId;
   delete output.documentId;
   output.id = documentId;
   return output;
 };
 
-export { switchIdForDocumentId };
+export const transformFieldsOrSort = (input: string[] | string): string[] | string => {
+  if (Array.isArray(input)) {
+    return input.map((item) => (item === 'id' ? 'documentId' : item));
+  }
+  return input.replace('id', 'documentId');
+};
+
+// TODO what if UID is undefined? throw?
```

Comment:
Is there any case where a uid could be empty? If not I would leave things as they are, and the typings should warn us if using this api incorrectly